recipes/uchardet/all/conanfile.py

```diff
@@ -0,0 +1,83 @@
+from conans import ConanFile, CMake, tools
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class UchardetConan(ConanFile):
+    name = "uchardet"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/freedesktop/uchardet"
+    description = "uchardet is an encoding detector library, which takes a sequence of bytes in an unknown character encoding and attempts to determine the encoding of the text. Returned encoding names are iconv-compatible."
+    topics = ("conan", "encoding", "detector")
+    license = "MPL 1.1"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "check_sse2": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "check_sse2": True,
+    }
+
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _patch_sources(self):
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                              "${CMAKE_BINARY_DIR}/uchardet.pc",
+                              "${CMAKE_CURRENT_BINARY_DIR}/uchardet.pc")
```

Comment:
Not 100% sure, but to be more robust, you can just replace all uses of `CMAKE_BINARY_DIR` with `CMAKE_CURRENT_BINARY_DIR`.
```suggestion
            "${CMAKE_BINARY_DIR}",
            "${CMAKE_CURRENT_BINARY_DIR}")
```

Reply:
makes sense 👍 I also created a PR on their repo https://github.com/freedesktop/uchardet/pull/1